src/utils/userService.js

```diff
@@ -0,0 +1,108 @@
+import { authHeader } from "./utils";
+
+const login = async (email, password) => {
+  const requestOptions = {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify({ email, password }),
+  };
+
+  const response = await fetch(`/users/authenticate`, requestOptions);
+
+  const user = await handleResponse(response);
+  localStorage.setItem("user", JSON.stringify(user));
+  return user;
+};
+const notes = async (pageNumber) => {
+  const requestOptions = {
+    method: "GET",
+    headers: authHeader(),
+    params: { date: "", page: pageNumber },
+  };
+  return fetch(`/notes`, requestOptions).then((response) =>
+    JSON.parse(response.text())
+  );
+};
+const sharedNotes = async (pageNumber) => {
+  const requestOptions = {
+    method: "GET",
+    headers: authHeader(),
+    params: { page: pageNumber },
+  };
+  return fetch(`/sharednotes/get`, requestOptions).then((response) =>
+    JSON.parse(response.text())
+  );
+};
+const logout = () => {
+  localStorage.removeItem("user");
+};
+
+const getAll = () => {
+  const requestOptions = {
+    method: "GET",
+    headers: authHeader(),
+  };
+
+  return fetch(`/users`, requestOptions).then((response) =>{
+    return JSON.parse(response.text())}
+  );
+};
+
+const register = (user) => {
+  const requestOptions = {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify(user),
+  };
+
+  return fetch(`/users/register`, requestOptions).then(handleResponse);
+};
+
+const setSharedNotes = (sharedNotes)=>{
+  const requestOptions = {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
```

Comment:
And all the headers too. You can try to create a common request sender to avoid declaration of the structure each time